jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/panel/PanelGridGeneralTester.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2000-2016 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.test.showcase.panel;
+
+import org.junit.Test;
+
+import com.liferay.faces.test.selenium.Browser;
+import com.liferay.faces.test.selenium.assertion.SeleniumAssert;
+import com.liferay.faces.test.showcase.TesterBase;
+
+
+/**
+ * @author  Kyle Stiemann
+ * @author  Philip White
+ */
+public class PanelGridGeneralTester extends TesterBase {
+
+	@Test
+	public void runPanelGridGeneralTest() throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, "panelGrid", "general");
+
+		// Test the visibility of text elements in every row and column of the table.
+		SeleniumAssert.assertElementVisible(browser,
+			"(//div[contains(@class,'showcase-example-usage')]/table//td[contains(.,'Row 1 Col 1')])");
```

Comment:
This seems like a good place to use `SeleniumAssert.assertElementTextVisible()` to ensure that each cell has the correct text. In fact, adding a method to do this would probably also make this test more readable. Something like `assertRowColTextCorrect(Browser browser, int row, int col)`.